### PR TITLE
Code clean up

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.ApiExplorer/DefaultApiDescriptionProvider.cs [hunk 1]
```diff
@@ -579,7 +576,7 @@ public PseudoModelBindingVisitor(ApiParameterContext context, ParameterDescripto
                 Context = context;
                 Parameter = parameter;
 
-                Visited = new HashSet<PropertyKey>();
+                Visited = new HashSet<PropertyKey>(new PropertyKeyEqualityComparer());
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.ApiExplorer/DefaultApiDescriptionProvider.cs]: The comparer wasn't set although it seems expressly designed for it.

File: src/Microsoft.AspNetCore.Mvc.ApiExplorer/DefaultApiDescriptionProvider.cs [hunk 2]
```diff
@@ -736,7 +733,11 @@ public bool Equals(PropertyKey x, PropertyKey y)
 
                 public int GetHashCode(PropertyKey obj)
                 {
-                    return obj.ContainerType.GetHashCode() ^ obj.PropertyName.GetHashCode() ^ obj.Source.GetHashCode();
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.ApiExplorer/DefaultApiDescriptionProvider.cs]: this null refs. Just using a HashCodeCombiner instead.

File: src/Microsoft.AspNetCore.Mvc.Core/Formatters/TextOutputFormatter.cs [hunk 3]
```diff
@@ -218,13 +217,8 @@ private IList<StringWithQualityHeaderValue> Sort(IList<StringWithQualityHeaderVa
                 }
                 else if (value.Quality != null)
                 {
-                    count++;
                     sortNeeded = true;
                 }
-                else
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/Formatters/TextOutputFormatter.cs]: doesn't look like there's any significance to `count` or this branch.

File: src/Microsoft.AspNetCore.Mvc.Core/Internal/ResourceInvoker.cs [hunk 4]
```diff
@@ -766,7 +763,7 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
 
                         _diagnosticSource.AfterOnResultExecuting(resultExecutingContext, filter);
 
-                        if (_resultExecutingContext.Cancel == true)
+                        if (_resultExecutingContext.Cancel)
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/Internal/ResourceInvoker.cs]: George Boole is disappoint.

File: src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/CollectionModelBinder.cs [hunk 5]
```diff
@@ -355,32 +354,6 @@ protected virtual void CopyToModel(object target, IEnumerable<TElement> sourceCo
             }
         }
 
-        private static object[] RawValueToObjectArray(object rawValue)
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/CollectionModelBinder.cs]: unused

File: src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Internal/ModelBindingHelper.cs [hunk 6]
```diff
@@ -275,7 +275,7 @@ public static class ModelBindingHelper
                 valueProvider,
                 modelMetadata,
                 bindingInfo: null,
-                modelName: prefix ?? string.Empty);
+                modelName: prefix);
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Internal/ModelBindingHelper.cs]: We null check `prefix`

File: src/Microsoft.AspNetCore.Mvc.Cors/DependencyInjection/MvcCorsMvcCoreBuilderExtensions.cs [hunk 7]
```diff
@@ -38,12 +38,8 @@ public static IMvcCoreBuilder AddCors(this IMvcCoreBuilder builder)
             }
 
             AddCorsServices(builder.Services);
-
-            if (setupAction != null)
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Cors/DependencyInjection/MvcCorsMvcCoreBuilderExtensions.cs]: We null check `setupAction`

File: src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs [hunk 8]
```diff
@@ -284,10 +274,6 @@ public override void EnsureRenderedBodyOrSections()
             }
         }
 
-        // Working around an issue with ApiCheck tool
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Razor/RazorPage.cs]: Seems to be fixed

File: src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelMetadata.cs [hunk 9]
```diff
@@ -419,7 +419,7 @@ public bool Equals(ModelMetadata other)
         /// <inheritdoc />
         public override bool Equals(object obj)
         {
-            return base.Equals(obj as ModelMetadata);
+            return Equals(obj as ModelMetadata);
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Abstractions/ModelBinding/ModelMetadata.cs]: Presumably we want to call the `Equals(ModelMetadata)` overload.

File: src/Microsoft.AspNetCore.Mvc.Core/Formatters/TextOutputFormatter.cs (hunk 3 above)
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/Formatters/TextOutputFormatter.cs]: It tells you the count

File: src/Microsoft.AspNetCore.Mvc.Formatters.Json/Internal/JsonResultExecutor.cs [hunk 10]
```diff
@@ -99,8 +99,8 @@ public virtual Task ExecuteAsync(ActionContext context, JsonResult result)
 
             var response = context.HttpContext.Response;
 
-            string resolvedContentType = null;
-            Encoding resolvedContentTypeEncoding = null;
+            string resolvedContentType;
+            Encoding resolvedContentTypeEncoding;
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Formatters.Json/Internal/JsonResultExecutor.cs]: `out var`?

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Formatters.Json/Internal/JsonResultExecutor.cs]: Yup. I wanna do another PR with C# 7 shenanigans.
